src/runtime/dict.cpp

```diff
@@ -32,6 +32,24 @@ using pyston::ExceptionStyle::ExceptionStyle;
 
 Box* dictRepr(BoxedDict* self) {
     std::vector<char> chars;
+    int status = Py_ReprEnter((PyObject*)self);
+    if (status != 0) {
+        try {
+            if (status < 0)
+                throwCAPIException();
+        } catch (ExcInfo e) {
+            Py_ReprLeave((PyObject*)self);
+            throw e;
+        }
```

Comment:
I don't think we should have this try-catch block (will call Py_ReprLeave when we shouldn't).